config/v1alpha1/tests/insightsdatagathers.config.openshift.io/InsightsConfig.yaml

```
@@ -14,3 +14,145 @@ tests:
+          gatherConfig:
+            storageSpec:
+              type: PersistentVolumeClaim
+      expectedError: 'spec.gatherConfig.storageSpec: Invalid value: "object": persistentVolume is required when type is PersistentVolumeClaim, and forbidden otherwise'
+    - name: When storageSpec.type is not PersistentVolumeClaim then PersistentVolume must not be present
+      initial: |
+        apiVersion: config.openshift.io/v1alpha1
+        kind: InsightsDataGather
+        spec:
+          gatherConfig:
+            storageSpec:
+              type: Ephemeral
+              persistentVolume:
+                persistentVolumeClaim:
+                  name: test-claim
+      expectedError: 'spec.gatherConfig.storageSpec: Invalid value: "object": persistentVolume is required when type is PersistentVolumeClaim, and forbidden otherwise'
+    - name: mountPath must not contain colon
+      initial: |
+        apiVersion: config.openshift.io/v1alpha1
+        kind: InsightsDataGather
+        spec:
+          gatherConfig:
+            storageSpec:
+              type: PersistentVolumeClaim
+              persistentVolume:
+                mountPath: /data:/data
+                persistentVolumeClaim:
+                  name: test-claim
+      expectedError: 'spec.gatherConfig.storageSpec.persistentVolume.mountPath: Invalid value: "string": mountPath must not contain a colon'
+    - name: invalid disabledGatherers name
+      initial: |
+        apiVersion: config.openshift.io/v1alpha1
+        kind: InsightsDataGather
+        spec:
+          gatherConfig:
+            disabledGatherers:
+              - invalid_gatherer_1
+      expectedError: 'Invalid value: "string": disabledGatherer must be in the format of {gatherer}/{function} where the gatherer and function are lowercase strings that may include underscores (_) and are separated by a forward slash (/) if the function is provided'
+    - name: storageSpec.type is required
+      initial: |
+        apiVersion: config.openshift.io/v1alpha1
+        kind: InsightsDataGather
+        spec:
+          gatherConfig:
+            storageSpec:
+              persistentVolume:
+                mountPath: /data
+                persistentVolumeClaim:
+                  name: test-claim
+      expectedError: "spec.gatherConfig.storageSpec.type: Required value"
+    - name: storageSpec.PersistenVolumeClaim can not be empty
+      initial: |
+        apiVersion: config.openshift.io/v1alpha1
+        kind: InsightsDataGather
+        spec:
+          gatherConfig:
+            storageSpec:
+              type: PersistentVolumeClaim
+              persistentVolume:
+                mountPath: /data
+                persistentVolumeClaim:
+      expectedError: 'spec.gatherConfig.storageSpec.persistentVolume.persistentVolumeClaim: Required value, <nil>: Invalid value: "null": some validation rules were not checked because the object was invalid; correct the existing errors to complete validation'
```

We care about the required value for this case in particular IIUC
```suggestion
      expectedError: spec.gatherConfig.storageSpec.persistentVolume.persistentVolumeClaim: Required value
```